src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -559,6 +559,51 @@ await invokeTask.NoThrowAwaitable(); // may or may not throw due to cancellation
         await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>(nameof(Server.TestInvalidMethod)));
     }
 
+    [Fact]
+    public async Task InvokeAsync_InvokesMethodWithAttributeSet()
+    {
+        string correctCasingResult = await this.clientRpc.InvokeWithParameterObjectAsync<string>("test/InvokeTestMethod");
+        Assert.Equal("test method attribute", correctCasingResult);
+
+        await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>("teST/InvokeTestmeTHod"));
+
+        string baseMethodResult = await this.clientRpc.InvokeAsync<string>("base/InvokeMethodWithAttribute");
+        Assert.Equal("base InvokeMethodWithAttribute", baseMethodResult);
+
+        await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>("base/InvokeVirtualMethodOverride"));
+
+        await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>("child/InvokeVirtualMethodOverride"));
```

Comment:
Added explanation.  The reason why we throw conflict for all cases is because we we do the method invocation, we always invoke the method in the derived method.  If there are conflicts, you can argue that as long as we pass in a method name that's the same as the derived method MethodNameAttribute, then we should invoke that method and not throw an exception.  But what happens if you pass in the base method MethodNameAttribute?  You would expect it to invoke the base method, but it doesn't.  For simplicity (also for the sake of I can't see a reason why you would ever want conflicts), I made it throw as soon as a conflict is detected.